cmd/agent/app/options/options.go

```diff
@@ -119,6 +131,7 @@ func (o *Options) AddFlags(fs *pflag.FlagSet, allControllers []string) {
 	fs.DurationVar(&o.ResyncPeriod.Duration, "resync-period", 0, "Base frequency the informers are resynced.")
 	fs.IntVar(&o.ConcurrentClusterSyncs, "concurrent-cluster-syncs", 5, "The number of Clusters that are allowed to sync concurrently.")
 	fs.IntVar(&o.ConcurrentWorkSyncs, "concurrent-work-syncs", 5, "The number of Works that are allowed to sync concurrently.")
+	fs.BoolVar(&o.EnableReportSecret, "enable-report-secret", true, "Enable report secret when registering cluster.")
```

Comment:
We have two secrets now on a Cluster, one is for accessing the member cluster(`.spec.SecretRef`) and the other one is for an impersonator(`.spec.ImpersonatorSecretRef`).
So, we have to distinguish between the two secrets.

My suggestion is:
```suggestion
fs.StringSliceVar(&o.ReportSecrets, "report-secrets", []string{"KubeCredentials", "KubeImpersonator"}, "The secrets that are allowed to be reported to the Karmada control plane during registering. Valid values are 'KubeCredentials', 'KubeImpersonator' and 'None'. e.g 'KubeCredentials,KubeImpersonator' or 'None'.")
```
What do you think? 
also, cc @XiShanYongYe-Chang here for the impersonator thing, tasks need to do after this flag.

Reply:
cc @Garrybest who might need to know this.

Reply:
+1, I agree with this flag.

Reply:
should  flag `report-secrets` add to karmada-controller-manager too ?

Reply:
I don't think so. 
`karmada-controller-manager` won't report secret.